src/UUID.hs

```diff
@@ -14,11 +14,10 @@ import Control.Monad.State
 type UUID = Int
 
 -- | Computes the successor 'UUID' of the current 'UUID'.
-next :: State UUID ()
+next :: State UUID UUID
```

Comment:
Why changing it from `State UUID ()` to `State UUID UUID`?

Reply:
Because the usage of `next` always looked like
```
do
  next
  id <- get
```

With this change the generated id will be returned directly:
```
do
  id <- next
```